core/unix/include/sigcontext.h

```diff
@@ -328,6 +328,28 @@ typedef struct _kernel_sigcontext_t {
     unsigned char __reserved[4096] __attribute__((__aligned__(16)));
 } kernel_sigcontext_t;
 
-#endif /* __aarch64__ */
+#    ifndef FPSIMD_MAGIC
+/*
+ * Header to be used at the beginning of structures extending the user
+ * context. Such structures must be placed after the rt_sigframe on the stack
+ * and be 16-byte aligned. The last structure must be a dummy one with the
+ * magic and size set to 0.
+ */
+struct _aarch64_ctx {
+    __u32 magic;
+    __u32 size;
+};
+
+#        define FPSIMD_MAGIC 0x46508001
+
+struct fpsimd_context {
```

Comment:
What happens is that if you build for aarch64 (host==target) it uses the signal.h for aarch64 and pulls this definition in.  If you do host!=target on amd64 host, it has nowhere to find this struct.  Maybe we should rename its use in our kernel_sigcontex_t so it's always coming from here but I wanted to minimize the change.  I'll add a comment.

Reply:
Wouldn't it be better to have an isolated header file and extract that code out of signal.h in order to avoid duplication?

Reply:
@derekbruening I think you missed this suggestion. Not to worry, it was a nit anyways.

Reply:
Yes, we have tried to get rid of including the system signal.h before but IIRC there were an awful lot of defines coming from there and we balked at having full copies, so the current state is that we have differently-named types like "kernel_sigcontext_t".  I agree it would be better to have a full copy and eliminate signal.h from core/ building.  I feel like this came up in multiple issues in the past.  If it's not in the issue tracker already maybe we can file a new one.

Reply:
Filed #4338.